File: 이건희/SWexpertacademy/D3/1221.py
T = int(input())
wordChange = {"ZRO" : 0, "ONE" : 1, "TWO" : 2, "THR" : 3, "FOR" : 4, "FIV" : 5, "SIX" : 6, "SVN" : 7, "EGT" : 8, "NIN" : 9}
numChange = {0 : "ZRO", 1 : "ONE", 2 : "TWO", 3 : "THR", 4 : "FOR", 5 : "FIV", 6 : "SIX", 7 : "SVN",  8 : "EGT", 9 : "NIN"}
 
for _ in range(1, T + 1):
    test_case, N = map(str, input().split())
    nums = list(input().split())
     
    for i in range(int(N)):
        nums[i] = wordChange[nums[i]]
         
    nums.sort()
     
    for i in range(int(N)):
        nums[i] = numChange[nums[i]]
         
    print(f'{test_case}')
    print(' '.join(nums))

File: 이훈일/삼성 SW 역량 테스트 기출 문제/D3/1221.java
import java.util.Scanner;
import java.io.FileInputStream;
import java.util.*;

class Solution {
    static HashMap<String, Integer> map = new HashMap<String, Integer>() {{
    	put("ZRO", 0);
        put("ONE", 1);
        put("TWO", 2);
        put("THR", 3);
        put("FOR", 4);
        put("FIV", 5);
        put("SIX", 6);
        put("SVN", 7);
        put("EGT", 8);
        put("NIN", 9);
    }};
    
	public static void main(String args[]) throws Exception {
		Scanner sc = new Scanner(System.in);
		int T;
		T=sc.nextInt();
		
		for(int test_case = 1; test_case <= T; test_case++) {
			String prefix = sc.next();
            int N = sc.nextInt();
            String[] nums = new String[N];
            for(int i = 0; i < N; i++) {
            	nums[i] = sc.next();
            }
            
            Arrays.sort(nums, new Comparator<String>(){
            	@Override
                public int compare(String o1, String o2) {
                	return map.get(o1) - map.get(o2);
                }
            });
            
            StringBuilder sb = new StringBuilder();
            for(String num : nums) {
            	sb.append(" " + num); 
            }
            
            System.out.println(prefix + sb.toString());
		}
	}
}
